src/VisualStudio/Core/Def/Implementation/InheritanceMargin/InheritanceMarginHelpers.cs

```diff
@@ -51,5 +56,74 @@ public static ImageMoniker GetMoniker(InheritanceRelationship inheritanceRelatio
             // The relationship is None. Don't know what image should be shown, throws
             throw ExceptionUtilities.UnexpectedValue(inheritanceRelationship);
         }
+
+        public static ImmutableArray<InheritanceMenuItemViewModel> CreateMenuItemViewModelsForSingleMember(ImmutableArray<InheritanceTargetItem> targets)
+        {
+            var targetsByRelationship = targets.OrderBy(target => target.DisplayName).GroupBy(target => target.RelationToMember).ToImmutableArray();
+            if (targetsByRelationship.Length == 1)
+            {
+                // If all targets have one relationship.
+                // e.g. interface IBar { void Bar(); }
+                // class A : IBar { void Bar() {} }
+                // class B : IBar { void Bar() {} }
+                // for 'IBar', the margin would be I↓. So header is not needed.
+                return targetsByRelationship[0].SelectAsArray(target => TargetMenuItemViewModel.Create(target, indent: false)).CastArray<InheritanceMenuItemViewModel>();
+            }
+            else
+            {
+                // Otherwise, it means these targets has different relationship,
+                // these targets would be shown in group, and a header should be shown as the first item to indicate the relationship to user.
+                using var _ = CodeAnalysis.PooledObjects.ArrayBuilder<InheritanceMenuItemViewModel>.GetInstance(out var builder);
+                foreach (var (relationship, targetItems) in targetsByRelationship)
+                {
+                    builder.AddRange(CreateMenuItemsWithHeader(relationship, targetItems));
+                }
+
+                return builder.ToImmutable();
```

Comment:
```suggestion
return targetsByRelationShip.SelectMany(pair => CreateMenuItemsWithHeader(pair.Item1, pair.Item2));                
```

I'm not sure if this ends up being better or worse than using an ArrayBuild. @CyrusNajmabadi  do you know?